perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AttributesManagerBlImpl.java

```diff
@@ -6058,6 +6088,50 @@ public void checkAttributeDependencies(PerunSession sess, RichAttribute richAttr
 	}
 
 	/**
+	 * Checks if the attributes represent empty values. If so, converts them into null.
+	 *
+	 * @param attributes attributes to be checked if are null
+	 */
+	public void convertEmptyAttrValueToNull(List<Attribute> attributes) throws ConsistencyErrorException {
+		for (Attribute attribute : attributes) {
+			convertEmptyAttrValueToNull(attribute);
+		}
+	}
+
+	/**
+	 * Checks if the attribute represents empty value. If so, converts it into null.
+	 *
+	 * @param attribute attribute to be checked if is null
+	 */
+	public void convertEmptyAttrValueToNull(Attribute attribute) throws ConsistencyErrorException {
+		if(attribute.getValue() == null) {
+			return;
+		}
+		if (attribute.getValue() instanceof Integer) {
+			return;
+		}
+		if (attribute.getValue() instanceof String) {
+			if(((String)attribute.getValue()).matches("\\s*")) {
+				attribute.setValue(null);
+			}
+		} else if (attribute.getValue() instanceof Boolean) {
+			if(attribute.getValue().equals(Boolean.FALSE)) {
+				attribute.setValue(null);
+			}
+		} else if (attribute.getValue() instanceof ArrayList) {
+			if(((ArrayList)attribute.getValue()).isEmpty()) {
+				attribute.setValue(null);
+			}
+		} else if (attribute.getValue() instanceof LinkedHashMap) {
+			if(((LinkedHashMap)attribute.getValue()).isEmpty()) {
+				attribute.setValue(null);
+			}
+		} else {
+			throw new ConsistencyErrorException("Invalid attribute value type: " + attribute.getType());
```

Comment:
Please include also actual value java class to the exception message, since our data types doesn't necessarily match it (LargeString, LargeArray for example).